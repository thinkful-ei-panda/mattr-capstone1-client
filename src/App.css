.App {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

a, h1, h2, p {
  color: #000055;
}

a{
  margin: 10px;
  text-align: center;
  align-self: center;
  text-decoration: none;
  margin-bottom: 10px;
}

img {
  width: 25%;
}

div.Nav{
  text-align: center;
}

section {
  background-color: #960000;
}

main {
  width: 50%;
  margin-right: auto;
  margin-left: auto;
  background-color: #ffffffe0;
  padding: 20px;
 
}

@media screen and (min-width: 480px) {
  main {
    width: 80%;
    /* background-color: lightgreen; */
  }

  img {
    width: 50%;
  }
 
  .scorecard-entry {
    display: flex;
  }
}