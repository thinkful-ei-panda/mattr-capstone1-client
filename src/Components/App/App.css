 .App {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 0;
  
}

body {
  font-family: 'Kanit', sans-serif;
  background: linear-gradient(rgb(20, 30, 48), rgb(36, 59, 85),rgb(20, 30, 48));
  background-size: cover;
}

a, h1 {
  color: white;
  text-align: center;
  margin-top: 10px;
  padding: 0;
  margin-bottom: 0;
  font-family: 'Kanit', sans-serif;
}

h2 {
  color: white;
  padding-bottom: 0;
  margin-bottom: 0;
}

h2, h3, h4 {
  text-align: center;
}

h3 {
  color: #03e9f4
}

h4 {
  color: #243b55;
  font-size: 1.3rem;
  margin-top: 5px;
  margin-bottom: 5px;
}

p {
  margin-top: 5px;
  margin-bottom: 5px;
  align-self: center;
}

a {
  margin: 10px;
  text-align: center;
  align-self: center;
  text-decoration: none;
  margin-bottom: 10px;
  font-family: 'Kanit', sans-serif;
  font-size: .7em;
  color: #03e9f4
}

img {
  width: 90%;
  margin-top: 5px;
  box-shadow: 2px 2px 12px 3px #03e9f4;
}

.Nav{
  text-align: center;
  font-size: 1.75rem;
}

.scorecard-entry {
margin-top: 10px;
display: flex;
width: 100%;
}

.stats {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  background-color: #ffffff;
  
}

label {
  color: #000055;
  margin-top: 5px;
}

.vote-link {
display: flex;
align-content: center;
justify-content: center;
margin: 20px;
font-size: 1.5rem;
font-family: 'Kanit', sans-serif;
}


form {
  color: #ffffffe0;
  width: 100%;
  margin-right: auto;
  margin-left: auto;
}

input[type=radio] {
  font-size: 1.2em;
  height: 1em;
  width: 1em;
  border: 0;
  padding:10px;
  border:0;
  border-radius:10px;
  
}

input[type=text] {
padding: 10px;
margin:10px;
}

button.Button.vote-link {
  width: 90%;
}

p.red {
  color: red;
}
  
.login-box, .vote-box {
  margin: 0;
  flex-direction: row;
  text-align: center;
  background: rgba(0,0,0,.5);
  box-sizing: border-box;
  box-shadow: 0 15px 25px rgba(0,0,0,.6);
  border-radius: 10px;
  justify-content: center;
  width: 100%;
}

.login-box h2 {
  margin-top: 20px;
  padding: 0;
  color: #fff;
  text-align: center;
}

.login-box .user-box {
  display: flex;
}

.login-box .user-box input {
  width: 100%;
  padding: 10px 0;
  font-size: 16px;
  color: #fff;
  margin-bottom: 30px;
  border: none;
  border-bottom: 1px solid #fff;
  outline: none;
  background: transparent;
}
.login-box .user-box label {
  padding: 10px 0;
  font-size: 16px;
  color: #fff;
  pointer-events: none;
  transition: .5s;
}

.login-box .user-box input:focus ~ label,
.login-box .user-box input:valid ~ label {
  top: -20px;
  left: 0;
  color: #03e9f4;
  font-size: 12px;
}

.login-box form a, .vote-link {
  display: flex;
  flex-direction: column;
  padding: 10px 20px;
  color: #03e9f4;
  font-size: 16px;
  text-decoration: none;
  text-transform: uppercase;
  overflow: hidden;
  transition: .5s;
  margin-top: 40px;
  letter-spacing: 4px
}

.login-box a:hover , .vote-link:hover {
  background: #03e9f4;
  color: #fff;
  border-radius: 5px;
  box-shadow: 0 0 5px #03e9f4,
              0 0 25px #03e9f4,
              0 0 50px #03e9f4,
              0 0 100px #03e9f4;
}

.login-box a span, .vote-link span {
  position: absolute;
  display: block;
}

.login-box a span:nth-child(1), .vote-link span:nth-child(1)  {
  top: 0;
  left: -100%;
  width: 100%;
  height: 2px;
  background: linear-gradient(90deg, transparent, #03e9f4);
  animation: btn-anim1 1s linear infinite;
}

@keyframes btn-anim1 {
  0% {
    left: -100%;
  }
  50%,100% {
    left: 100%;
  }
}

.login-box a span:nth-child(2), .vote-link span:nth-child(2) {
  top: -100%;
  right: 0;
  width: 2px;
  height: 100%;
  background: linear-gradient(180deg, transparent, #03e9f4);
  animation: btn-anim2 1s linear infinite;
  animation-delay: .25s
}

@keyframes btn-anim2 {
  0% {
    top: -100%;
  }
  50%,100% {
    top: 100%;
  }
}

.login-box a span:nth-child(3), .vote-link span:nth-child(3) {
  bottom: 0;
  right: -100%;
  width: 100%;
  height: 2px;
  background: linear-gradient(270deg, transparent, #03e9f4);
  animation: btn-anim3 1s linear infinite;
  animation-delay: .5s
}

@keyframes btn-anim3 {
  0% {
    right: -100%;
  }
  50%,100% {
    right: 100%;
  }
}

.login-box a span:nth-child(4), .vote-link span:nth-child(4) {
  bottom: -100%;
  left: 0;
  width: 2px;
  height: 100%;
  background: linear-gradient(360deg, transparent, #03e9f4);
  animation: btn-anim4 1s linear infinite;
  animation-delay: .75s
}

@keyframes btn-anim4 {
  0% {
    bottom: -100%;
  }
  50%,100% {
    bottom: 100%;
  }
}

button.vote-link {
  min-height: 10px;
  margin: 0;
  padding: 0;
  background-color: #141e3007;
  padding: 10px;
  border: none;
}

.vote-link-button-group {
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
}

.user-box{
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
}

.not-logged {
  display: flex;
  flex-direction: column;
}

    @media screen and (min-width: 1200px) {
      img {
        width: 90%;
        margin-top: 5px;
      }
    
      .scorecard-entry {
        display: flex;
        flex-direction: column;
        width: 100%;
        margin: 10px;

      }

      .election-cards {
        display: flex;
        flex-direction: row;
        justify-content: center;
      }

      a{
        font-size: 0.8em;
      }

      h4.light {
        font-size: 2.5em;
        margin-top: 5px;
        margin-bottom: 5px;

      }

      p.light {
        font-size: 2.2em;
        margin-top: 5px;
        margin-bottom: 5px;
      }

      .login-box, .vote-box {
        margin: 0;
        padding: 20px;
        flex-direction: row;
        text-align: center;
        width: 80%;
        margin-right: auto;
        margin-left: auto;
        background: rgba(0,0,0,.5);
        box-sizing: border-box;
        box-shadow: 0 15px 25px rgba(0,0,0,.6);
        border-radius: 10px;
        text-align: center;
      }

      div.stats {
        width: 100%;
        display: flex;
        align-items: center;
        justify-content: center;
        flex-direction: column;
        background-color: #ffffff;
        
      }

      
    } 

