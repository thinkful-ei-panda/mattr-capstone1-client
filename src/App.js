import React, { Component } from 'react'
import { Route, Switch } from 'react-router-dom'

import './App.css';
import ScoreCard from './ScoreCard';
import VoteScreen from './VoteScreen';
import Header from './Header'
import Nav from './Nav'
import Democrat from './joe-biden.jpg'
import Republican from './donald-trump.jpeg'
import NotLoggedInScreen from './NotLoggedInScreen';
// import UserRegistrationScreen from './UserRegistrationScreen';
import RegistrationForm from './RegistrationForm/RegistrationForm'
import UserLoginScreen from './UserLoginScreen';
import VoteConfirmationScreen from './VoteConfirmationScreen';
import NotFoundPage from './NotFoundPage';
// import { BrowserRouter, Route, Link } from "react-router-dom";



class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      users: [
        {
          id: '',
          firstName: '',
          lastName: '',
          DOB: '',
          email: '',
          userName: '',
          votePresident: '',
          loggedIn: false,

        }
      ],
      voteScreen: false,
      demCount: 0,
      repCount: 0,
      error: null,
    }
  }

 

  handleSubmit(e) {
    e.preventDefault();
    const baseUrl = 'http://localhost:8000/users/';
    const params = [];
    if (this.state.id) {
      params.push(`search=${this.state.search}`);
    }
    
    const query = params.join('&');
    const url = `${baseUrl}?${query}`;

    fetch(url)
      .then(res => {
        if (!res.ok) {
          throw new Error(res.statusText);
        }
        return res.json();
      })
      .then(data => {
        this.setState({
          voteScreen: true,
          error: null
        });
      })
      .catch(err => {
        this.setState({
          error: 'Sorry, could not get books at this time.'
        });
      })

  }

  render() {
    
    
    return (
      <main className="App">
        <Header />
        <Nav />
        <ScoreCard />
         {/* <HomePage /> */}
          {/* <Route path='/' component={HomePage} /> */}
          <Switch>
          <Route path='/ScoreCard' component={ScoreCard} />
          <Route path='/VoteScreen' component={VoteScreen} />
          <Route path='/NotLoggedInScreen' component={NotLoggedInScreen} />
          <Route path='/RegistrationForm' component={RegistrationForm} />
          <Route path='/UserLoginScreen' component={UserLoginScreen} />
          <Route path='/VoteConfirmationScreen' component={VoteConfirmationScreen} />
          <Route component={NotFoundPage} />
          </Switch>
       
      </main>
    );
  }
}

export default App;