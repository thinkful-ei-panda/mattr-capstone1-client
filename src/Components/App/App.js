import React, { Component } from 'react'
import { Route, Switch, Redirect } from 'react-router-dom'

import PrivateRoute from '../Utils/PrivateRoute'
import './App.css';
import Election from '../Election/Election';

import Header from '../Header'
import Nav from '../Nav'
import Home from '../Home'
import NotLoggedIn from '../NotLoggedIn';
import RegistrationForm from '../RegistrationForm/RegistrationForm'
import Login from '../Login/Login';
import Vote from '../Vote/Vote'
import VoteConfirmation from '../VoteConfirmation';
import NotFoundPage from '../NotFoundPage';



export default class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      users: [
        {
          id: '',
          firstName: '',
          lastName: '',
          DOB: '',
          email: '',
          userName: '',
          votePresident: '',
          loggedIn: false,

        }
      ],
      voteScreen: false,
      demCount: 0,
      repCount: 0,
      error: null,
    }
  }

  handleSubmit(e) {
    e.preventDefault();
    const baseUrl = 'http://localhost:8000/user/';
    // const params = [];  
    // const query = params.join('&');
    // const url = `${baseUrl}?${query}`;
    const url = `${baseUrl}`;
    fetch(url)
      .then(res => {
        if (!res.ok) {
          throw new Error(res.statusText);
        }
        return res.json();
      })
      .then(data => {
        this.setState({
          error: null
        });
      })
      .catch(err => {
        this.setState({
          error: 'Sorry, could not complete operation.'
        });
      })

  }

  render() {
    
    return (
      <main className="App">
        <Header />
        <Nav />        
        <Switch>
        <Route exact path="/"
                render={() => {
                    return (
                      this.state.isUserAuthenticated ?
                      <Redirect to="/home" /> :
                      <Redirect to="/NotLoggedIn" /> 
                    )
                }}
              />
               <Route exact path="/Home" component={Home} />
        <Route path='/Election' component={Election} />
        <PrivateRoute path='/Vote' component={Vote} />
        <Route path='/NotLoggedIn' component={NotLoggedIn} />
        <Route path='/RegistrationForm' component={RegistrationForm} />
        <Route path='/Login' component={Login} />
        <PrivateRoute path='/VoteConfirmation' component={VoteConfirmation} />
        <Route component={NotFoundPage} />
        </Switch>
       
      </main>
    );
  }
}

