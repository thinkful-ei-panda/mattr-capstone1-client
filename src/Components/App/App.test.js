import App from "./App";
import React from "react";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import history from '../../history'

import { Router } from "react-router-dom";

describe(`App Component`, () => {
    describe(`Smoke test`, () => {
        it("should render without crashing", () => {
            const div = document.createElement('div');
            ReactDOM.render(
              <Router history={history}>
                <App />
              </Router>,
              div );
              ReactDOM.unmountComponentAtNode(div);
          });

    })

    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <Router history = {history}>
                        <App />
                    </Router>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });

})


