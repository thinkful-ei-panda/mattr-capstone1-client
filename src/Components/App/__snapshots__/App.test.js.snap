// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot test Renders the UI as expected 1`] = `
<main
  className="App"
>
  <div>
    <h1>
      My Poll Data
    </h1>
  </div>
  <div>
    <div
      className="Nav"
    >
      <a
        href="/Home"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/Election"
        onClick={[Function]}
      >
        Elections
      </a>
      <a
        href="/RegistrationForm"
        onClick={[Function]}
      >
         
        Register
         
      </a>
      <a
        href="/Login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
    <div
      className="login-box"
    >
      <p>
         Welcome to My Poll Data. You are now entering the beta site. Let us begin our journey into shaping the future of Democracy. As we know the current process for registering and voting is time consuming, cumbersome and in some ways enigmatic for Billions of people around the globe. My Poll Data aims to leverage technology to make Democracy more efficient and therefore more representative. 
      </p>
      <p>
         Use your email to register and login to test the voting system out. Alternatively, you can use this demo account: 
      </p>
      <p>
         login email: mycoolemail@gmail.com 
      </p>
      <p>
         password: My7$ecret 
      </p>
      <a
        className="vote-link"
        href="/RegistrationForm"
        onClick={[Function]}
      >
         Register 
      </a>
    </div>
  </div>
</main>
`;
